python/lsst/display/matplotlib/matplotlib.py

```diff
@@ -275,22 +319,41 @@ def _i_mtv(self, data, wcs, title, isMask):
 
                 colors[i + 1][3] = alpha
 
-            dataArr = maskArr
             cmap = mpColors.ListedColormap(colors)
             norm = mpColors.NoNorm()
         else:
-            cmap = pyplot.cm.gray
+            cmap = self._image_colormap
             norm = self._normalize
 
         ax = self._figure.gca()
         bbox = data.getBBox()
-        mappable = ax.imshow(dataArr, origin='lower', interpolation='nearest',
-                             extent=(bbox.getBeginX() - 0.5, bbox.getEndX() - 0.5,
-                                     bbox.getBeginY() - 0.5, bbox.getEndY() - 0.5),
-                             cmap=cmap, norm=norm)
+        extent = (bbox.getBeginX() - 0.5, bbox.getEndX() - 0.5,
+                  bbox.getBeginY() - 0.5, bbox.getEndY() - 0.5)
+
+        with pyplot.rc_context(dict(interactive=False)):
+            if isMask:
+                for i, p in reversed(list(enumerate(planeList))):
+                    if colors[i + 1][3] == 0:
```

Comment:
Explain for the reader of the code
1. Why `i+1` instead of `i`?
2. That `[3]` is the alpha channel.  Perhaps as simple as 
```
alpha_channel = 3
[...]
     if colors[i + 1][alpha_channel] == 0
```

Could do these even higher up above so that the `colors[i + 1][3] = alpha` in line 320 used this same `alpha_channel`.  It's more obvious in line 320 because of the variable name on the rhs, but for consistency could do both.